lib/csv/writer.rb

```diff
@@ -94,7 +94,7 @@ def prepare_header
       return unless @headers
 
       converter = @options[:header_fields_converter]
-      @headers = converter.convert(@headers, nil, 0)
+      @headers = converter.convert(@headers, nil, 0, [])
```

Comment:
We need to care about a case that `headers` is the following `String`:

```csv
"quoted",unquoted,"quoted-again"
```

Could you also add a test for this case?